### PR TITLE
Synch editing add ons

### PR DESCRIPTION
**Pull request by ViditChitkara**
Some bug fixes related to #805

### CODE REVIEW

**Comment by ViditChitkara** [comment]
Hi, So there are a few bugs here which I realised after testing the recently updated development branch. We have a few fixes here. Kindly have a look @alaxalves @jywarren @sashadev-sky

**Comment by ViditChitkara** [comment]
Previously if we added or deleted an image in a map, then it wouldn't reflect in others who are connected to the map. With this, it'll work now. Also, previously we had a single channel for all the maps which would have be conflicting changes from various maps into a single map. So, a separate channel is streamed for each map. This way, all changes would be mutually independent with respect to each map. 
Apart from this a few bug fixes are made.

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -92,10 +92,7 @@ def update
       @warpable.locked = params[:locked]
       @warpable.cm_per_pixel = @warpable.get_cm_per_pixel
       @warpable.save
-      respond_to do |format|
-        format.html { render html: 'success' }
-        format.json { render json: @warpable.map.fetch_map_data }
-      end
+      render json: @warpable.map.fetch_map_data
```
Comment (alaxalves) [inline app/controllers/images_controller.rb]: I solved this same thing. I guess it came from a merge request.

**Comment by ViditChitkara** [comment]
Fixed the rubocop offences here!! Is this good to go now? @jywarren

**Comment by jywarren** [comment]
Does this enable any of the new features or is it just preparatory? If it does, can you share a gif? Thank you!

**Comment by ViditChitkara** [comment]
No, not an additional feature, but a few things to note:-
1. Creates a separate channel for each map which would else have caused interference.
2. A few bug fixes in #805 which may have come at the time of merging with development
3. Earlier an added image or deleted image wasn't reflected in the other connected browser. Added that as well.

Will share a gif.

**Comment by ViditChitkara** [comment]
gsoc-gif-1

**Comment by ViditChitkara** [comment]
gsoc-gif-2

**Comment by ViditChitkara** [comment]
@jywarren here are the gifs

**Comment by jywarren** [comment]
Wow! And, with the 'revisions' shown in the images menu now, can we update that to show who last edited the image?

**Comment by ViditChitkara** [comment]
> Wow! And, with the 'revisions' shown in the images menu now, can we update that to show who last edited the image?

Umm--I need to confirm that. Although, we can do that I think. Will see that and make a PR for it.
Thanks

**Pull request merged**